core/unix/signal.c

```diff
@@ -2692,11 +2697,16 @@ thread_set_self_context(void *cxt)
                         sizeof(frame.uc.uc_sigmask));
     LOG(THREAD_GET, LOG_ASYNCH, 2, "thread_set_self_context: pc=" PFX "\n", sc->SC_XIP);
     LOG(THREAD_GET, LOG_ASYNCH, 3, "full sigcontext\n");
-    DOLOG(LOG_ASYNCH, 3,
-          { dump_sigcontext(dcontext, get_sigcontext_from_rt_frame(&frame)); });
+    DOLOG(LOG_ASYNCH, 3, {
+        dcontext_t *dcontext = get_thread_private_dcontext();
```

Comment:
This is shadowing the var of the same name in the outer scope: please fix.  Surprising the compiler doesn't warn about it (MSVC complains about such things).  Maybe best to just put it back how it was, with dcontext in all builds.  Your goal was to remove the call to gtpd for arm?  Is it not removed by the compiler in a release build?

Reply:
All usages of `dcontext` in this function are conditional (on either some environment or build type) now, unlike before this PR. So it's slightly tricky to avoid unused var errors.

For this case, it seems that `DOLOG` is defined only under a complex combination (`INTERNAL && DEBUG && !STANDALONE_DECODER`). So I figured it's best to let there be a separate declaration here; I'll just change the identifier so that it doesn't shadow the one in outer scope.

Reply:
Done.